packages/ui-components/src/components/NativeSelectOptionGroup/NativeSelectOptionGroup.component.tsx

```diff
@@ -0,0 +1,50 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Juno contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from "react"
+
+export interface NativeSelectOptionGroupProps extends React.OptgroupHTMLAttributes<HTMLOptGroupElement> {
+  /**
+   * A visible label for the option group.
+   * It will be displayed as the group heading in the dropdown.
+   */
+  label?: string
+
+  /**
+   * Determines whether the option group is disabled.
+   * If true, all options within the group will also be disabled.
+   */
+  disabled?: boolean
+
+  /**
+   * Additional CSS classes to apply to the option group for custom styling.
+   */
+  className?: string
+
+  /**
+   * Elements to be rendered within the option group.
+   * This can be any React node or a collection of React nodes.
+   * Typically, these should be NativeSelectOption components.
+   */
+  children?: React.ReactNode
```

Comment:
Should you do here maybe same as for combobox PR like restrict element with a generic props??

Reply:
Didn't get around to it, thanks for catching that. I really like the approach if we expect specific children. 💪

Reply:
Reverted this change in case it breaks things for how apps are using children now. 
Improvement added to [epic](https://github.com/cloudoperators/juno/issues/570) under "For some elements, we could make sure children are only elements (and props) we expect, if parent only works with those children".

Reply:
ok